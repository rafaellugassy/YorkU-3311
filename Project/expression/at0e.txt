equals

start_of_set_enumeration
addition
int_value(1)
int_value(3)
addition
int_value(2)
int_value(2)
end_of_set_enumeration

start_of_set_enumeration
int_value(4)
end_of_set_enumeration

analyze
-- Before comparing for equality, first simply the LHS and RHS sets
simplify
