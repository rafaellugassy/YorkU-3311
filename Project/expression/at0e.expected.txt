  Expression currently specified: ?
  Report: Expression is initialized.
->equals
  Expression currently specified: (? = nil)
  Report: OK.
->start_of_set_enumeration
  Expression currently specified: ({?} = nil)
  Report: OK.
->addition
  Expression currently specified: ({(? + nil)} = nil)
  Report: OK.
->int_value(1)
  Expression currently specified: ({(1 + ?)} = nil)
  Report: OK.
->int_value(3)
  Expression currently specified: ({(1 + 3), ?} = nil)
  Report: OK.
->addition
  Expression currently specified: ({(1 + 3), (? + nil)} = nil)
  Report: OK.
->int_value(2)
  Expression currently specified: ({(1 + 3), (2 + ?)} = nil)
  Report: OK.
->int_value(2)
  Expression currently specified: ({(1 + 3), (2 + 2), ?} = nil)
  Report: OK.
->end_of_set_enumeration
  Expression currently specified: ({(1 + 3), (2 + 2)} = ?)
  Report: OK.
->start_of_set_enumeration
  Expression currently specified: ({(1 + 3), (2 + 2)} = {?})
  Report: OK.
->int_value(4)
  Expression currently specified: ({(1 + 3), (2 + 2)} = {4, ?})
  Report: OK.
->end_of_set_enumeration
  Expression currently specified: ({(1 + 3), (2 + 2)} = {4})
  Report: OK.
->analyze
  Expression currently specified: ({(1 + 3), (2 + 2)} = {4})
  Report: ({(1 + 3), (2 + 2)} = {4}) is type-correct.
->simplify
  Expression currently specified: ({(1 + 3), (2 + 2)} = {4})
  Report: True
